packages/big-design/src/components/TableNext/Row/Row.tsx

```diff
@@ -1,60 +1,165 @@
+    index: number,
+    childrenRows: T[],
+    isParentRow: boolean,
+    isExpandable: boolean,
+  ): void;
 }
 
 const InternalRow = <T extends TableItem>({
+  childrenRows,
+  childRowIndex,
   columns,
   forwardedRef,
   headerCellWidths,
+  isDraggable,
   isDragging = false,
+  isExpandable = false,
   isSelectable = false,
   isSelected = false,
   item,
+  parentRowIndex,
   showDragIcon = false,
   onItemSelect,
+  onExpandedRow,
+  isExpanded = false,
+  selectedItems,
+  isParentRow = false,
   ...rest
 }: RowProps<T> & PrivateProps) => {
-  const onChange = () => {
-    if (onItemSelect) {
-      onItemSelect(item);
+  const { hasChildrenRows, label, onChange, onExpandedChange, isChecked, isIndeterminate } =
+    useRowState({
+      childRowIndex,
+      childrenRows,
+      isExpandable,
+      isParentRow,
+      isSelected,
+      onExpandedRow,
+      onItemSelect,
+      selectedItems,
+      parentRowIndex,
+    });
+
+  const renderSelectDataCell = () => {
+    if (isSelectable && isParentRow) {
+      return (
+        <DataCell
+          cellPadding={0}
+          isCheckbox={true}
+          isExpandable={isExpandable}
+          key="data-checkbox"
+          width={10}
+          withHorizontalPadding={false}
+          withVerticalPadding={false}
+        >
+          <Checkbox
+            checked={isChecked}
+            hiddenLabel
+            isIndeterminate={isIndeterminate}
+            label={label}
+            onChange={onChange}
+            width={0}
+          />
+        </DataCell>
+      );
     }
-  };
 
-  const label = isSelected ? `Selected` : `Unselected`;
+    return null;
+  };
 
-  return (
-    <StyledTableRow isDragging={isDragging} isSelected={isSelected} ref={forwardedRef} {...rest}>
-      {showDragIcon && (
+  const renderDragIconCell = () => {
+    if (showDragIcon && isParentRow) {
+      return (
         <DataCell width={headerCellWidths[0]}>
           <DragIndicatorIcon />
         </DataCell>
-      )}
-      {isSelectable && (
-        <DataCell isCheckbox={true} key="data-checkbox">
-          <Checkbox checked={isSelected} hiddenLabel label={label} onChange={onChange} />
+      );
+    }
+
+    return null;
+  };
+
+  const renderExpandedIconCell = () => {
+    if (isExpandable && isParentRow && hasChildrenRows) {
+      const needsHorizontalPadding = !isSelectable && !isDraggable;
+
+      return (
+        <DataCell withHorizontalPadding={needsHorizontalPadding} withVerticalPadding={false}>
+          <StyledExpandedIcon onClick={onExpandedChange}>
+            {isExpanded ? <ExpandMoreIcon /> : <ChevronRightIcon />}
+          </StyledExpandedIcon>
         </DataCell>
-      )}
+      );
+    }
+
+    return null;
+  };
+
+  const renderExtraCellsForParentRow = () => {
+    const extraDataCells: ReactNode[] = [];
+
+    if (!hasChildrenRows) {
+      extraDataCells.push(<DataCell key={`parent-extra-cell-${parentRowIndex}-1`} />);
+    }
+
+    return extraDataCells;
+  };
```

Comment:
We could simplify this a bit instead of relying on array mutations
```suggestion
  const renderExtraCellsForParentRow = (): ReactNode[] => {
    if (!hasChildrenRows) {
      return [<DataCell key={`parent-extra-cell-${parentRowIndex}-1`} />];
    }

    return [];
  };
```